data/transactions/logic/eval.go

```diff
@@ -2133,6 +2163,38 @@ func opGtxna(cx *EvalContext) {
 	cx.stack = append(cx.stack, sv)
 }
 
+func opGtxnas(cx *EvalContext) {
+	last := len(cx.stack) - 1
+
+	gtxid := int(cx.program[cx.pc+1])
+	if gtxid >= len(cx.TxnGroup) {
+		cx.err = fmt.Errorf("gtxnas lookup TxnGroup[%d] but it only has %d", gtxid, len(cx.TxnGroup))
+		return
+	} else if gtxid < 0 {
+		cx.err = fmt.Errorf("gtxnas lookup %d cannot be negative", gtxid)
+		return
+	}
```

Comment:
I'll fix this up when I do loads, stores, but just want to point out that we don't want to complain that the number is negative.  In AVM, it isn't.  It's only negative (potentially) because this code casted a big uint64 to an int, and that can give you a negative result.

Reply:
Makes sense!

Reply:
PS, I liked the way you handled this in args a lot.  There, you used uint64 mostly, and casted the len() calls instead of the stack argument.  I think I will fix this stuff up that way, instead of the way I had been contemplating.